### PR TITLE
chore: Spelling

### PR DESCRIPTION
**Pull request by jsoref**
**What this PR does / why we need it**:
Follow-up to #7258

**Special notes for your reviewer**:
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/helm/commit/b2fab15593c7b6cc5f3e727b7174d415dae86fa0#commitcomment-47528550

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/helm/commit/5c3b078e3c3fba78320b828eb50bb604acd31ed8

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

As usual, I'm happy to drop things.

This PR is fancier than the previous PR as it's built around an action that enables projects to prevent new typos from being merged (instead of relying on me making periodic PRs). At the very least, I'll maintain a branch so that I can record which [corrections not to suggest](
https://github.com/jsoref/helm/blob/b2fab15593c7b6cc5f3e727b7174d415dae86fa0/.github/actions/spelling/patterns.txt#L81-L89.
) going forward.

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: CONTRIBUTING.md
Comment (jsoref) [inline CONTRIBUTING.md:262]: brand

File: cmd/helm/root_unix.go
Comment (jsoref) [inline cmd/helm/root_unix.go:30]: Yes, I know some people think the former is a thing. Neither [books](https://books.google.com/ngrams/graph?content=panicking%2Cpanicing&year_start=1800&year_end=2019&corpus=26&smoothing=3&direct_url=t1%3B%2Cpanicking%3B%2Cc0%3B.t1%3B%2Cpanicing%3B%2Cc0) nor the [Internet](https://books.google.com/ngrams/graph?content=panicking%2Cpanicing&year_start=1800&year_end=2019&corpus=26&smoothing=3&direct_url=t1%3B%2Cpanicking%3B%2Cc0%3B.t1%3B%2Cpanicing%3B%2Cc0) agree.

File: cmd/helm/testdata/testcharts/chart-with-template-lib-dep/charts/common/templates/_ingress.yaml
Comment (jsoref) [inline cmd/helm/testdata/testcharts/chart-with-template-lib-dep/charts/common/templates/_ingress.yaml:7]: Since this is a test chart, its content shouldn't matter, that said, it isn't a problem for me to drop this (or anything else).

File: pkg/action/upgrade.go
Comment (jsoref) [inline pkg/action/upgrade.go:97]: brand

File: pkg/chartutil/dependencies_test.go
Comment (jsoref) [inline pkg/chartutil/dependencies_test.go:193]: [Jabberwocky by Lewis Carroll](https://www.poetryfoundation.org/poems/42916/jabberwocky)

File: pkg/chartutil/save_test.go
Comment (jsoref) [inline pkg/chartutil/save_test.go:105]: This is a user visible change

File: pkg/engine/lookup_func.go
Comment (jsoref) [inline pkg/engine/lookup_func.go:102]: It's likely this is an API break which I'll need to drop. This function appears to be new (d564d4bf2d6c6a9781439db231c4aaa1d7a916f8) since the previous round: #6752 (v3.1.0?)

File: pkg/kube/client.go
Comment (jsoref) [inline pkg/kube/client.go:108]: Yes, it's cute. No, I don't think it adds value. Happy to drop.

File: pkg/storage/driver/sql.go
Comment (jsoref) [inline pkg/storage/driver/sql.go:291]: User facing

File: testdata/releases.yaml
Comment (jsoref) [inline testdata/releases.yaml:20]: Three Musketeers is a historical adventure novel written in 1844 by French author Alexandre Dumas ... D'Artagnan offers the letter to Athos, Porthos, and Aramis

File: pkg/engine/lookup_func.go
Comment (bacongobbler) [reply pkg/engine/lookup_func.go:102]: it's a private function. Not an API break.

File: pkg/kube/client.go
Comment (bacongobbler) [reply pkg/kube/client.go:108]: I would strongly disagree that it adds value. Please revert.

File: pkg/action/upgrade.go
Comment (bacongobbler) [inline pkg/action/upgrade.go:409]: this was intentional.

https://www.youtube.com/watch?v=YCGeeCIzBs4

File: pkg/chartutil/save_test.go
Comment (bacongobbler) [reply pkg/chartutil/save_test.go:105]: it's just the output from a failing test, so it isn't a user visible change.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: pkg/storage/driver/sql.go
Comment (bacongobbler) [reply pkg/storage/driver/sql.go:291]: Changing error messages to fix typos are OK.

File: pkg/action/upgrade.go
Comment (jsoref) [reply pkg/action/upgrade.go:409]: Ah. Yeah, I'm familiar with that Americana.

**Review by hickeyma — APPROVED** [review]
Who could have thought there are so many spelling mistakes lurking. Thanks for this @jsoref. Requires a second approval for maintainer for merging.

**Comment by jsoref** [comment]
@hickeyma: this is basically only one year's worth of new mistakes since I already cleaned up the project at the end of 2019...

If you're interested in the action, I'd be more than happy to contribute it.

**Review by jdolitsky — APPROVED** [review]

**Pull request merged**
